doc/RPC.md

```diff
@@ -0,0 +1,170 @@
+RPC
+===
+
+Remote Procedure Calls are used within Sia to trigger actiosn between server and client.
+Most RPCs are send to all connected peers of a gateway using a broadcast call,
+while some RPCs have a Client and Server version,
+to provide bidirectional one-to-one communication.
+
+The ID of the RPC operation is sent first as a `[8]byte` header,
+which can be found as the `rpcID` type in the Golang gateway's RPC code.
+IDs that are longer than 8 bytes, are truncated to fit within this limit.
+
+After that the Request Arguments are sent.
+
+The Reply content is sent as last, for those (one-to-one) RPCs that have replies.
+
+All message content is encoded using the Encoding discussed in the Encoding.md document.
+Please read up on that document for more information about the specific layout of
+each RPC operation's package content.
+
+All RPC Operations return errors,
+some of which get logged in Debug mode, others which just get ignored.
+
+### RPC Operations
+
+#### ShareNodes
+
+Used as a one-to-one operation.
+
+Registered in the `gateway` module.
+
+Request Package:
+
+```
+[ []byte{"ShareNod"} | []string{nodesNetAddresses} ]
```

Comment:
We should model these after the API docs. Something like:

>### Call Listing
>
>Unless otherwise, specified, these calls follow a request/response pattern and use the [encoding](./Encoding.md) package to serialize data.
>
>**All data received via RPC should be considered untrusted and potentially malicious.**
>
>#### ShareNodes
>
>ShareNodes requests peer addresses from a peer. The gateway calls this RPC regularly to update its list of potential peers.
>
>ID: `"ShareNod"`
>
>Request: None
>
>Response:
> 
>```go
>[]modules.NetAddress
>```
>
>Recommendations:
>
>- Requesting peers should limit the response to 3000 bytes.
>- Responding peers should send no more than 10 peers, and should not send peers that are unlikely to be reachable.